doc: update mention of generating bitcoin.conf

epiccurious opened the pull request:
Closes #30153.

This PR updates `doc/init.md` to mention generating an example bitcoin.conf instead of referencing the placeholder `share/examples/bitcoin.conf`. Also changes the code-formatted text to a markdown link.

## Background

- Two years ago, `share/examples/bitcoin.conf` was replaced with [a placeholder file](https://github.com/bitcoin/bitcoin/commit/b483084d866c16d97a34251ae652bac94f85f61d). To see an example `bitcoin.conf`, the user now runs the `contrib/devtools/gen-bitcoin-conf.sh` script, which replaces the placeholder file with the parsed contents of `bitcoind --help`.

- The instructions in `init.md` about an example `bitcoin.conf` haven't changed significantly since they were [added almost 10 years ago](https://github.com/bitcoin/bitcoin/blame/234bfbf6a5fcba37e510e9cb6c1f2a629cd0290e/doc/init.md#L39). They should be updated to improve clarity.

epiccurious committed:
Link to gen-bitcoin-conf.sh instead of bitcoin.conf placeholder

edilmedeiros commented on share/examples/bitcoin.conf:
```diff
@@ -1 +1 @@
-# This is a placeholder file. Please follow the instructions in `contrib/devtools/README.md` to generate a bitcoin.conf file.
+# This is a placeholder file. Please follow the instructions in the `gen-bitcoin-conf.sh` section of `contrib/devtools/README.md` to generate a bitcoin.conf file.
```
I don't like this change that much since this hardens the logical link between this file and `contrib/devtools/README.md` without any kind of automation to check if e.g. the section `gen-bitcoin-conf.sh` still exist.

epiccurious replied:
Good point. My intention here is to reduce ambiguity by being more explicit to the user.

One solution would be to add a check to ensure this section mentioned actually exists in the markdown, but it's probably not worth the added code complexity for little benefit.

Do you prefer to revert this change back?

epiccurious replied:
Agreed. Resolved with aef43a3a996618d581e0a16f7fc5e842708e83c9.

edilmedeiros reviewed (commented):
Good catch.

I thought this would not make sense for people getting the binaries, but seems that none of the binary distributions include the `doc` folder. IIUIC, this change affects only people who is getting core from source. If so, the update does make sense.

BenWestgate commented:
Concept ACK

epiccurious commented:
> IIUIC, this change affects only people who is getting core from source

Correct as I understand it. Only impacts people using the source, not the binary distributions.

kevkevinpal commented:
lgtm ACK [42f788d](https://github.com/bitcoin/bitcoin/pull/30154/commits/42f788d071095b62783c53a75f06831855e86bbd)

The `share/examples/bitcoin.conf` files links to `contrib/devtools/README.md` so instead of send the users to a file to link to another file it makes sense just to use that link directly here

davidgumberg commented:
You should squash this into a single commit and update the PR description

epiccurious commented:
> squash this into a single commit and update the PR description

Squashed into 9013e2b97e8f50d2be63ce740c42d0b0e0b9b7f2 and updated PR description.

edilmedeiros commented:
ACK 9013e2b97e8f50d2be63ce740c42d0b0e0b9b7f2

stickies-v reviewed (approved):
ACK 9013e2b97e8f50d2be63ce740c42d0b0e0b9b7f2

meta nit: for trivial changes, discussion can happen on the PR directly, no need to open a separate issue first

kevkevinpal commented:
reACK [9013e2b](https://github.com/bitcoin/bitcoin/pull/30154/commits/9013e2b97e8f50d2be63ce740c42d0b0e0b9b7f2)

tdb3 reviewed (commented):
ACK for 9013e2b97e8f50d2be63ce740c42d0b0e0b9b7f2
Good find.  Makes sense to point the user to conf file generation (helping to ensure the conf reflects the implementation).  Followed the link (seemed to work fine).

achow101 commented:
ACK 9013e2b97e8f50d2be63ce740c42d0b0e0b9b7f2
